{"name": "test_div[TypeError]", "status": "failed", "statusDetails": {"message": "AssertionError: assert \"unsupported ...nt' and 'str'\" == \"TypeError: u...nt' and 'str'\"\n  - TypeError: unsupported operand type(s) for /: 'int' and 'str'\n  ? -----------\n  + unsupported operand type(s) for /: 'int' and 'str'", "trace": "self = <testing.test_calculator.TestCalculator object at 0x04510A48>, c = 100, d = '-', divexpect = 0\n\n    @allure.story(\"除法运算测试\")\n    @pytest.mark.parametrize('c,d,divexpect', get_divdatas_calc()[0], ids=get_divdatas_calc()[1])\n    @pytest.mark.div\n    def test_div(self, c, d, divexpect):\n        try:\n            # calc = calculator()\n            # assert 1 == self.calc.div(1, 1)\n>           res = self.calc.div(c, d)\n\ntesting\\test_calculator.py:51: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <pythoncode.calculator.calculator object at 0x04510358>, a = 100, b = '-'\n\n    def div(self, a, b):\n>       return a / b\nE       TypeError: unsupported operand type(s) for /: 'int' and 'str'\n\npythoncode\\calculator.py:9: TypeError\n\nDuring handling of the above exception, another exception occurred:\n\nself = <testing.test_calculator.TestCalculator object at 0x04510A48>, c = 100, d = '-', divexpect = 0\n\n    @allure.story(\"除法运算测试\")\n    @pytest.mark.parametrize('c,d,divexpect', get_divdatas_calc()[0], ids=get_divdatas_calc()[1])\n    @pytest.mark.div\n    def test_div(self, c, d, divexpect):\n        try:\n            # calc = calculator()\n            # assert 1 == self.calc.div(1, 1)\n            res = self.calc.div(c, d)\n        except Exception as e:\n            print(\"发生异常，异常信息{}，进行异常断言\".format(e))\n            # assert str(e) == 'division by zero'\n    \n>           assert str(e) == f\"TypeError: unsupported operand type(s) for /: 'int' and 'str'\"\nE           assert \"unsupported ...nt' and 'str'\" == \"TypeError: u...nt' and 'str'\"\nE             - TypeError: unsupported operand type(s) for /: 'int' and 'str'\nE             ? -----------\nE             + unsupported operand type(s) for /: 'int' and 'str'\n\ntesting\\test_calculator.py:56: AssertionError"}, "parameters": [{"name": "c", "value": "100"}, {"name": "d", "value": "'-'"}, {"name": "divexpect", "value": "0"}], "start": 1624035759645, "stop": 1624035759646, "uuid": "10602ade-6d48-4782-9220-565d6c535f75", "historyId": "c146aeb635abe4485f715f309bb44686", "testCaseId": "00f43c99e69edabe02f02364c544f882", "fullName": "test_calculator.TestCalculator#test_div", "labels": [{"name": "story", "value": "除法运算测试"}, {"name": "feature", "value": "加法和除法测试"}, {"name": "tag", "value": "div"}, {"name": "suite", "value": "test_calculator"}, {"name": "subSuite", "value": "TestCalculator"}, {"name": "host", "value": "QKFZ-888"}, {"name": "thread", "value": "9072-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test_calculator"}]}